src/libraries/Common/src/Extensions/ProviderAliasUtilities/ProviderAliasUtilities.cs

```diff
@@ -13,16 +14,24 @@ internal static class ProviderAliasUtilities
 
         internal static string GetAlias(Type providerType)
         {
-            foreach (CustomAttributeData attributeData in CustomAttributeData.GetCustomAttributes(providerType))
+            IList<CustomAttributeData> attributes = CustomAttributeData.GetCustomAttributes(providerType);
+
+            for (int i = 0; i < attributes.Count; i++)
             {
+                CustomAttributeData attributeData = attributes[i];
                 if (attributeData.AttributeType.FullName == AliasAttibuteTypeFullName)
                 {
-                    foreach (CustomAttributeTypedArgument arg in attributeData.ConstructorArguments)
+                    if (attributeData.ConstructorArguments.Count == 0)
                     {
-                        Debug.Assert(arg.ArgumentType == typeof(string));
-
-                        return arg.Value?.ToString();
+                        // Maybe the user defined their own ProviderAliasAttribute?
+                        continue;
                     }
```

Comment:
Nit: it looks a little weird to my eye that we nest for the outer if check but then inverted continue for the inner... I'd have expected something more like:
```C#
for (int i = 0; i < attributes.Count; i++)
{
    CustomAttributeData attributeData = attributes[i];
    if (attributeData.AttributeType.FullName != AliasAttibuteTypeFullName ||
        attributeData.ConstructorArguments.Count == 0) // Maybe the user defined their own ProviderAliasAttribute?
    {
        continue;
    }

    CustomAttributeTypedArgument arg = attributeData.ConstructorArguments[0];
    Debug.Assert(arg.ArgumentType == typeof(string));
    return arg.Value?.ToString();
}
```
Not a big deal, though.

Reply:
I think it's because I had the inner loop initially.